tooling-server-configuration/src/main/java/org/jboss/eap/qe/microprofile/tooling/server/logwatch/ModelNodeLogWatcher.java

```diff
@@ -0,0 +1,74 @@
+package org.jboss.eap.qe.microprofile.tooling.server.logwatch;
+
+import org.jboss.dmr.ModelNode;
+import org.wildfly.extras.creaper.core.online.ModelNodeResult;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * Log Watcher based on accessing messages through {@link org.jboss.dmr.ModelNode} returned from {@code :read-log-file}
+ * operation.
+ */
+public final class ModelNodeLogWatcher implements LogWatcher {
```

Comment:
I am against the idea of adding such constructor. As it was said before - logs can get pretty big and you don't want to get all that in the memory. The only option is to have constructor which would read some default amount of lines.

Reply:
OK. Can you add to the JavaDoc `-1` option which should get all the lines? And a test? 

I see what you are saying, however, IMHO it is not a good idea to limit developers. This might encourage them to think about it.

Reply:
+1 - I'll add it. In my opinion, some restriction makes sense. User (developer) must be aware of fact, that some amount of lines is being read into the memory which can have consequences. Mindlessly calling constructor with least amount of parameters could lead to some memory issues...